libparsec/crates/client/src/certif/user_update_profile.rs

```diff
@@ -0,0 +1,179 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_client_connection::ConnectionError;
+use libparsec_protocol::authenticated_cmds;
+use libparsec_types::prelude::*;
+
+use super::{
+    greater_timestamp, store::CertifStoreError, CertificateBasedActionOutcome, CertificateOps,
+    GreaterTimestampOffset, InvalidCertificateError, InvalidKeysBundleError,
+};
+use crate::EventTooMuchDriftWithServerClock;
+
+#[derive(Debug, thiserror::Error)]
+pub enum CertifUpdateUserProfileError {
```

Comment:
Note every variant here should lead to a corresponding test

(protip: you can copy `server_error`/`invalid_keys_bundle`/`invalid_response` tests from `libparsec/crates/client/tests/unit/certif/validate_child_manifest.rs` to handle the `InvalidKeysBundle` / `InvalidCertificate` / `Internal` variants)